pkg/scheduler/objects/application.go

```diff
@@ -1845,7 +1845,7 @@ func (sa *Application) RemoveAllAllocations() []*Allocation {
 		sa.appEvents.sendRemoveAllocationEvent(alloc, si.TerminationType_STOPPED_BY_RM)
 	}
 
-	if resources.IsZero(sa.pending) {
+	if resources.IsZero(sa.pending) && !resources.IsZero(resources.Add(sa.allocatedResource, sa.allocatedPlaceholder)) {
```

Comment:
why this condition check can fix that issue? or is it more suitable that we just check the existence of user tracker here?

Reply:
It fixes because we might not have an user tracker at all at this point if we call `RemoveApplication()`.

Reply:
@pbacsko thanks for response. I had another question:

why we don't check existence of tracker directly? for example: 

```golang
	if resources.IsZero(sa.pending) && ugm.GetUserManager().GetUserTracker(sa.user.User) != nil {
		sa.decUserResourceUsage(resources.Add(sa.allocatedResource, sa.allocatedPlaceholder), true)
	}
```

Reply:
That's a solution, too. Although I feel that in both cases, we need comments to explain why we do that because it's not obvious why we do it (one can definitely think: "what can possibly happen to the user tracker?"). I'm not against that change. But I don't think we need to return the tracker instance, just a boolean flag which indicates if it exists or not.

Another thing is that, IMO it makes sense to move this check to `decUserResourceUsage()`.

Reply:
> Another thing is that, IMO it makes sense to move this check to decUserResourceUsage().

that makes sense to me.

Reply:
My preference would be to use the user tracker retrieval. Overhead of the lookup including the lock is far less than doing the add and a compare to zero for the resource.